### ISSUE TITLE
[Bug]: Won't load model details after searching

### ISSUE THREAD

**Issue by drone540** [BlafKing/sd-civitai-browser-plus#210]
### Describe the bug.

After searching for anything in the browser. Clicking on anything results in nothing but errors on all boxes that would automatically fill in. Just updated to the latest version and the issue still persists.

### Steps to reproduce the problem.

1. Click on Search
2. Click any result
3. Errors out

### Expected behavior

Shows the model information and allows downloading

### System info

* Extension version: [84ca01ae](https://github.com/BlafKing/sd-civitai-browser-plus/commit/84ca01ae1e57cc955cf5cd40e42f2baf32cb26cb) 	2024-02-17 17:38:13
* OS: Kubuntu 22.04
* SD-WebUI version: [v1.6.0-2-g4afaaf8a](https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/4afaaf8a020c1df457bcf7250cb1c7f609699fa7)
* Python: 3.10.12

### Console logs

```Shell
Traceback (most recent call last):
  File "/home/kubuntu/Downloads/stable-diffusion-webui/venv/lib/python3.10/site-packages/gradio/routes.py", line 488, in run_predict
    output = await app.get_blocks().process_api(
  File "/home/kubuntu/Downloads/stable-diffusion-webui/venv/lib/python3.10/site-packages/gradio/blocks.py", line 1431, in process_api
    result = await self.call_function(
  File "/home/kubuntu/Downloads/stable-diffusion-webui/venv/lib/python3.10/site-packages/gradio/blocks.py", line 1103, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "/home/kubuntu/Downloads/stable-diffusion-webui/venv/lib/python3.10/site-packages/anyio/to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/home/kubuntu/Downloads/stable-diffusion-webui/venv/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "/home/kubuntu/Downloads/stable-diffusion-webui/venv/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "/home/kubuntu/Downloads/stable-diffusion-webui/venv/lib/python3.10/site-packages/gradio/utils.py", line 707, in wrapper
    response = f(*args, **kwargs)
  File "/home/kubuntu/Downloads/stable-diffusion-webui/extensions/sd-civitai-browser-plus/scripts/civitai_gui.py", line 428, in update_models_dropdown
    (html, tags, base_mdl, DwnButton, SaveImages, DelButton, filelist, filename, dl_url, id, current_sha256, install_path, sub_folder) = _api.update_model_info(model_string, model_versions.get('value'))
  File "/home/kubuntu/Downloads/stable-diffusion-webui/extensions/sd-civitai-browser-plus/scripts/civitai_api.py", line 760, in update_model_info
    meta = pic['meta']
KeyError: 'meta'
```

**Comment by victorbarlac** [comment]
Just had this problem too

**Comment by Abishai2007** [comment]
I'm having the same issue here.

**Comment by xvishon** [comment]
Also had this issue was working a few hours ago.

**Comment by LegendOriMedia** [comment]
Same issue -- used to work splendidly, now errors every time.  For clarity, the information doesn't even load.  The titles of the models show- but if I click on any, every field shows an error.
![image](https://github.com/BlafKing/sd-civitai-browser-plus/assets/116342158/54d3e038-11fb-4392-90f0-320a98668ead)

**Comment by hakaserver** [comment]
Probably broken since this civitai api commit from yesterday:
civitai/civitai@c88063a3d2caad93a62d5efa3a2ac7ce7dc77b87

**Comment by raydoomed** [comment]
the same problem

**Comment by hakaserver** [comment]
From what I understood with my limited knowledge, the API commit I mentioned above means that when asking Civitai's API for information through the model ID ('from model endpoint'), the response will not include the 'Meta' field anymore, which would include the prompt information for the model's sample gallery. In fact, when I manually requested the information from the [example in the Wiki](https://github.com/civitai/civitai/wiki/REST-API-Reference#example-3), the information retrieved doesn't contain 'Meta' like the example response had.

I think this means the extension will have to be updated to request the prompt info for each image in a different way. Last week @BlafKing mentioned he is away for vacation, so in the meantime a possible temporary fix could be to make the following changes:

In the sd-civitai-browser-plus\scripts folder, edit line 760 from civitai_api.py:
```
                    #meta = pic['meta'] change this line to
                    meta = pic['metadata']
```
I created a pull request just in case, but it's just a dirty fix make the extension work again, until a better fix is implemented.

**Comment by ResidentChief** [comment]
> From what I understand with my limited knowledge, the API commit I mentioned above means that when asking Civitai's API for information through the model ID ('from model endpoint'), the response will not include the 'Meta' field anymore, which would include the prompt information for the model's sample gallery. In fact, when I manually requested the information from the [example in the Wiki](https://github.com/civitai/civitai/wiki/REST-API-Reference#example-3), the information retrieved doesn't contain 'Meta' like the example response had.
> 
> I think this means the extension will have to be updated to request the prompt info for each image in a different way. Last week @BlafKing mentioned he is away for vacation, so in the meantime a possible temporary fix could be to make the following changes:
> 
> In the sd-civitai-browser-plus\scripts folder, edit line 760 from civitai_api.py:
> 
> ```
>                     #meta = pic['meta'] change this line to
>                     meta = pic['metadata']
> ```
> 
> I created a pull request just in case, but it's just a dirty fix make the extension work again, until a better fix is implemented.

Can confirm this fix works once restarting a1111/forge :)  Thanks heaps!!!

**Comment by drone540** [comment]
Did they mention why meta was removed in the first place? 

Maybe can open an issue there about the meta getting removed, and maybe have a way to request it?

**Comment by MysfitDev** [comment]
> Did they mention why meta was removed in the first place?
> 
> Maybe can open an issue there about the meta getting removed, and maybe have a way to request it?

I didn't see a reason why it was removed from the models API, but the meta object is still there in the /v1/model-versions API. I think it would just require pulling the data for the specific model version selected (or by hash for models that were already downloaded) and then fetching the prompts and other meta items to display from there.

**Comment by BlafKing** [comment]
Thanks for reporting everyone!

I've merged the temp fix from @ResidentChief for now, so that the main functions of the extension remain working, I'll start working on finding a way to retrieve the previous meta info again!

For now the latest commit should let you select models again.

**Comment by ResidentChief** [comment]
Wasn't my fix chief, but thanks for merging it :)

**Comment by BlafKing** [comment]
I've now included a proper fix in the latest version.

The extension now grabs every example image and it then extracts the prompt info from the metadata contained inside the image itself using the built-in functions from SD-WebUI.

This is a bit slower than it was before, since it needs to process each image individually now, but from my testing this was still faster than doing individual API requests per version/image, so that's why I went with this approach :,)
I'll be doing more testing in the future to find the best solution, but this should do for now!

Thanks again for reporting everyone, I'll go ahead and mark this issue as resolved now.
If you do end up experiencing more issues related to this, feel free to send another message here, to create new bug report!
